synapseclient/client.py

```diff
@@ -596,17 +534,49 @@ def get_client(cls, synapse_client: typing.Union[None, "Synapse"]) -> "Synapse":
             An instance of 'Synapse'.
 
         Raises:
-            SynapseError: No instance has been created - Please use login() first
+            SynapseError: No instance has been passed into a function where
+                the `synapse_client` kwarg is available, and no instance has been cached
+                during the Synapse constructor.
         """
         if synapse_client:
             return synapse_client
 
         if not cls._synapse_client:
             raise SynapseError(
-                "No instance has been created - Please use login() first"
+                "No instance has been passed into a function where the `synapse_client` kwarg is available, and no instance has been cached during the Synapse constructor."
```

Comment:
nit: The wording is a little confusing - I would re-write it as: 
> No Synapse client instance was provided, and no cached instance is available. Ensure that either an instance is passed into `synapse_client`, or a cached instance is available.

But this is subjective so I'll leave it up to you.